### DISCUSSION TITLE
Command to duplicate project

### DISCUSSION THREAD

**Discussion by MisterH001** [ddev/ddev#2997] [Ideas]
Thanks again for such an amazing tool. :)

Something that would be really (really) handy would be the ability to clone an entire project (including all files and database) in to a folder above the current folder (or the ability to specify a path).

This would be handy to quickly "spin" up a new version of the site in order to test a new feature or a bug, without affecting the site you're currently working on.

**Comment by rfay** [comment]
What a lot of people do for this kind of thing is use `ddev snapshot`. It makes a wonderful copy of the database. 

So for example, if you want to mess with your project, `ddev snapshot --name=beforemessing` then do whatever you want. Consider snapshotting the changed database. Then when you want to go back, `ddev snapshot restore`

**Reply by MisterH001** [reply]
Thanks @rfay 
Yes this is certainly one approach, but not exactly what I was suggesting.
I'll look to create a bash script to do something similar.
Thanks for your time :)

**Reply by rfay** [reply]
A bash script should work fine for you. See the [FAQ](https://ddev.readthedocs.io/en/stable/users/faq/) questions about "How do I move a project". Note that to do what you want to do the project name must change, because the project name must be unique on a given host computer.

**Reply by e-r-n-i-e** [reply]
> What a lot of people do for this kind of thing is use `ddev snapshot`. It makes a wonderful copy of the database.
> 
> So for example, if you want to mess with your project, `ddev snapshot --name=beforemessing` then do whatever you want. Consider snapshotting the changed database. Then when you want to go back, `ddev snapshot restore`

But this won't help if whatever I want to test involves changes to the files and/or changes to the ddev config. And sometimes it would be nice to have both versions active at the same time for example to compare the duplicated websites side by side.

What would be the neccessary steps to do this manually (as easy/fast as possible)?

**Comment by tyler36** [comment]
@e-r-n-i-e 

1. Clone the repo into another directly
2. Change/remove the `name` in the cloned repo's `.ddev/config.yaml`
3. Import the database

```
ddev snapshot restore [snapshot_name]
```

4. Install dependencies with `ddev composer install` and/or `ddev npm install`


Recently, I've begun dropping the `name` value in config completley when I commit the file. DDEV will use the folder name as your project name.

**Comment by gitressa** [comment]
> Recently, I've begun dropping the name value in config completley when I commit the file. DDEV will use the folder name as your project name.

That's a great tip @tyler36, thanks. I like a lot how minimal the `ddev config` command is. When I started using DDEV I found the [excellent documentation](https://ddev.readthedocs.io/en/latest/users/usage/commands/#config), where the example can hardly be any simpler, with only `docroot` and `project-type`. I was really impressed when I realized that it can be reduced further down to `ddev config --auto` and created #4963.
